### ISSUE TITLE
Android app crashing when opening streetview component.

### ISSUE THREAD

**Issue by WahabCcript** [nesterapp/react-native-streetview#47]
Hey, I am trying to integrate the streetview component in my react native app, it is working fine on iOS but seems to crash whenever I try to open it on android. I've tried to implement it from scratch to confirm whether or not I missed something but I'm still facing the same issue. 
Here are my project configurations:
package.json:
 "react-native": "0.70.1",
"react-native-reanimated": "^3.1.0",

android/build.gradle:
buildscript {
    ext {
        buildToolsVersion = "31.0.0"
        minSdkVersion = 21
        compileSdkVersion = 31
        targetSdkVersion = 31
        androidXCore = "1.6.0"  
        //kotlin_version = '1.6.10'

        if (System.properties['os.arch'] == "aarch64") {
            // For M1 Users we need to use the NDK 24 which added support for aarch64
            ndkVersion = "24.0.8215888"
        } else {
            // Otherwise we default to the side-by-side NDK version from AGP.
            ndkVersion = "21.4.7075529"
        }
        

I'm looking forward to your response.

**Comment by Windastella** [comment]
Same issue here.

**Comment by Windastella** [comment]
These are the log for the crash.

```java_vm_ext.cc:577] JNI DETECTED ERROR IN APPLICATION: can't call java.lang.String java.lang.Object.toString() on null object
java_vm_ext.cc:577]     in call to CallObjectMethodV
java_vm_ext.cc:577]     from void com.swmansion.reanimated.nativeProxy.EventHandler.receiveEvent(java.lang.String, com.facebook.react.bridge.WritableMap)```

Seems to be related to re animated. The last changes I did make before the crash happen is to update the Re-Animated Package from v2 to v3. Probably worth testing on your implementation. @ccriptdev7

**Comment by Windastella** [comment]
Downgraded my react-native-reanimated from v3 to v2. That solved the issue for now.

**Comment by ccriptdev7** [comment]
Yes exactly. I downgraded my react-native-reanimated to v2 which resolved the issue. However, downgrading react-native-reanimated caused problems in my iOS apps. I hope there's a path for this.

**Comment by mwh1r** [comment]
Am also experiencing this crash after upgrading `react-native-reanimated` to v3. Looks like there is some sort of conflict between reanimated and the `onSuccess`/`onError` events that this library emits. 

The way the react native [docs](https://reactnative.dev/docs/native-components-android#events) state to emit events differs from how this library is [doing it](https://github.com/nesterapp/react-native-streetview/blob/master/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetView.java#L73) and updating this library's code to the recommended approach happens to also bypass the problem.

I'll do a PR when I have a moment.

```patch
diff --git a/node_modules/react-native-streetview/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetView.java b/node_modules/react-native-streetview/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetView.java
index 369672c..2f0610a 100644
--- a/node_modules/react-native-streetview/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetView.java
+++ b/node_modules/react-native-streetview/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetView.java
@@ -13,8 +13,7 @@ import com.facebook.react.bridge.ReactContext;
 import com.facebook.react.bridge.ReadableMap;
 import com.facebook.react.bridge.WritableMap;
 import com.facebook.react.bridge.Arguments;
-import com.facebook.react.uimanager.UIManagerModule;
-import com.facebook.react.uimanager.events.EventDispatcher;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
 import com.google.android.gms.maps.OnStreetViewPanoramaReadyCallback;
 import com.google.android.gms.maps.StreetViewPanorama;
 import com.google.android.gms.maps.model.LatLng;
@@ -70,16 +69,15 @@ public class NSTStreetView extends StreetViewPanoramaView implements OnStreetVie
         this.panorama.setPanningGesturesEnabled(allGesturesEnabled);
         this.panorama.setStreetNamesEnabled(!streetNamesHidden);
 
-        final EventDispatcher eventDispatcher = ((ReactContext) getContext())
-                .getNativeModule(UIManagerModule.class).getEventDispatcher();
+        Context context = getContext();
+        ReactContext reactContext = (ReactContext)context;
 
         this.panorama.setOnStreetViewPanoramaCameraChangeListener(new StreetViewPanorama.OnStreetViewPanoramaCameraChangeListener() {
             @Override
             public void onStreetViewPanoramaCameraChange(StreetViewPanoramaCamera streetViewPanoramaCamera) {
                 if (!(streetViewPanoramaCamera.bearing >= 0 ) && coordinate != null) {
-                    eventDispatcher.dispatchEvent(
-                            new NSTStreetViewEvent(getId(), NSTStreetViewEvent.ON_ERROR)
-                    );
+                    reactContext.getJSModule(RCTEventEmitter.class)
+                            .receiveEvent(getId(), "onError", null);
                 }
             }
         });
@@ -91,13 +89,11 @@ public class NSTStreetView extends StreetViewPanoramaView implements OnStreetVie
                     WritableMap map = Arguments.createMap();
                     map.putDouble("latitude", streetViewPanoramaLocation.position.latitude);
                     map.putDouble("longitude", streetViewPanoramaLocation.position.longitude);
-                    eventDispatcher.dispatchEvent(
-                            new NSTStreetViewEvent(getId(), NSTStreetViewEvent.ON_SUCCESS, map)
-                    );
+                    reactContext.getJSModule(RCTEventEmitter.class)
+                            .receiveEvent(getId(), "onSuccess", map);
                 } else {
-                    eventDispatcher.dispatchEvent(
-                            new NSTStreetViewEvent(getId(), NSTStreetViewEvent.ON_ERROR)
-                    );
+                    reactContext.getJSModule(RCTEventEmitter.class)
+                            .receiveEvent(getId(), "onError", null);
                 }
             }
         });
diff --git a/node_modules/react-native-streetview/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetViewManager.java b/node_modules/react-native-streetview/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetViewManager.java
index 595b5c1..75a87b6 100644
--- a/node_modules/react-native-streetview/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetViewManager.java
+++ b/node_modules/react-native-streetview/android/src/main/java/co/il/nester/android/react/streetview/NSTStreetViewManager.java
@@ -55,9 +55,10 @@ public class NSTStreetViewManager extends SimpleViewManager<NSTStreetView> {
     @Override
     public @Nullable
     Map getExportedCustomDirectEventTypeConstants() {
-        return MapBuilder.of(
-                NSTStreetViewEvent.eventNameForType(NSTStreetViewEvent.ON_ERROR), MapBuilder.of("registrationName", NSTStreetViewEvent.eventNameForType(NSTStreetViewEvent.ON_ERROR)),
-                NSTStreetViewEvent.eventNameForType(NSTStreetViewEvent.ON_SUCCESS), MapBuilder.of("registrationName", NSTStreetViewEvent.eventNameForType(NSTStreetViewEvent.ON_SUCCESS))
-        );
+        return MapBuilder.builder()
+            .put("onSuccess",
+                    MapBuilder.of("registrationName", "onSuccess"))
+            .put("onError", MapBuilder.of("registrationName", "onError"))
+            .build();
     }
 }
```

**Comment by amitpdev** [comment]
fixed in v0.2.4
thanks for your work @mwh1r

**Comment by amitpdev** [comment]
fixed in v0.2.4
thanks for your work @mwh1r
